src/ProtoStruct.jl

```diff
@@ -1,10 +1,21 @@
+
+abstract type AbstractProto end
+
 macro proto( expr )
     if expr.head != Symbol("struct")
         throw(ArgumentError("Expected expression to be a type definition."))
     end
     ismutable = expr.args[1]
 
     name = expr.args[2]
+
+    if name.head == :<:
+        abstract_type = name.args[2]
+        name = name.args[1]
+    else
+        abstract_type = :(ProtoStructs.AbstractProto)
```

Comment:
is that good for anything or should we also just go `Any` here?

Reply:
Done!